jujugui/static/gui/src/app/jujulib/bakery-factory.js

```diff
@@ -0,0 +1,60 @@
+/* Copyright (C) 2016 Canonical Ltd. */
+'use strict';
+
+var module = module;
+
+(function (exports) {
+
+
+  var jujulib = exports.jujulib;
+  /**
+   * BakeryFactory to create and maintain bakery objects.
+   *
+   */
+
+  /**
+   Initialize.
+
+   @method initializer
+   @param {Object} cfg a config object containing the Bakery module
+                   {Bakery: Y.juju.environments.web.Bakery}
+   @return {undefined} Nothing.
+   */
+  function bakeryFactory(cfg) {
+    this.Bakery = cfg.Bakery;
+    this.bakeries = new Map();
+  };
+
+  bakeryFactory.prototype = {
+    /**
+     Return a bakery instance associated with the cfg.serviceName provided.
```

Comment:
I don't really like how `get` and `create` are handled by the same method. This will complicate the use a bit because if you go:

```JavaScript
bakeryfactory.get({serviceName: 'charmstore'});
```

This may or may not fail. If one has already been created it'll return it, if not. then it will fail to create a proper bakery instance. I'd much prefer that `create` and `get` are separate.